app/modal-windows/services/window.service.ts

```diff
@@ -0,0 +1,71 @@
+// File: window.service.ts
+// Created by: Daniel James
+// Date created: January 30, 2017
+//
+
+import { Injectable, NgZone } from '@angular/core';
+
+import { remote, ipcRenderer } from 'electron';
+import { ModalInfo, ModalService } from './../../models/modal-window'
+
+@Injectable()
+export class WindowService implements ModalService {
+    private callbacks = new Map<Number, (data: any) => void>();
+    public windowInfo: ModalInfo | null; // Main window will be null
+
+    constructor(private _ngZone: NgZone) {
+        ipcRenderer.on("windowResult", (event, arg) => this.callback(arg as ModalInfo));
+
+        // TODO: sending synchronous IPC here could be a bad idea, a better idea would be to send it async and store windowInfo as a Promise
+        this.windowInfo = ipcRenderer.sendSync("getWindowInfo", remote.getCurrentWindow().id);
+    }
+
+    public createModal(selector: string): [ModalInfo, Promise<any>] {
```

Comment:
Really my only question is: Do we support any other kind of child windows? Do we support secondary main windows? (Now or ever)